packages/asset-swapper/src/utils/market_operation_utils/index.ts

```diff
@@ -666,14 +667,30 @@ export class MarketOperationUtils {
             if (rfqt.isIndicative) {
                 // An indicative quote is being requested, and indicative quotes price-aware enabled
                 // Make the RFQT request and then re-run the sampler if new orders come back.
-                const indicativeQuotes = await rfqt.quoteRequestor.requestRfqtIndicativeQuotesAsync(
-                    makerToken,
-                    takerToken,
-                    amount,
-                    side,
-                    wholeOrderPrice,
-                    rfqt,
-                );
+
+                const indicativeQuotes =
+                    rfqt.rfqClient !== undefined
+                        ? ((
+                              await rfqt.rfqClient.fetchPricesAsync({
+                                  takerAddress: rfqt.takerAddress,
+                                  txOrigin: rfqt.txOrigin,
+                                  makerToken,
+                                  takerToken,
+                                  assetFillAmount: amount,
+                                  marketOperation: side,
+                                  comparisonPrice: wholeOrderPrice,
+                                  feeAmount: undefined,
+                                  feeToken: undefined,
+                              })
+                          ).quotes as V4RFQIndicativeQuoteMM[])
+                        : await rfqt.quoteRequestor.requestRfqtIndicativeQuotesAsync(
```

Comment:
is there a way we could make quoterequestor conform to IRfqClient so we don't have this weird branching? Or is it not worth it?

Reply:
I think it's better to have the IRfqClient conform to the existing code, otherwise, we need to change some stuff downstream. Riskier, for unclear benefit